sys/sys/pcpu.h

```diff
@@ -121,8 +124,9 @@ extern uintptr_t dpcpu_off[];
  * subtracted the DPCPU_START, so we do it now.
  */
 #define	_DPCPU_PTR(b, n)					\
-    (__typeof(DPCPU_NAME(n))*)((b) +                            \
-        ((vaddr_t)&DPCPU_NAME(n) - (vaddr_t)DPCPU_START))
+	cheri_setboundsexact((__typeof(DPCPU_NAME(n)) *)((b) +	\
+	    ((vaddr_t)&DPCPU_NAME(n) - (vaddr_t)DPCPU_START)),	\
+	    sizeof(DPCPU_NAME(n)))
```

Comment:
This needs to be CRRL'ed (which will be constant-folded, so no runtime overhead) or use cheri_setbounds. Former is preferable since it asserts the input is aligned.

Reply:
I wonder if we want a helper that does a CRRL of the passed in length vs doing it inline in places that use `cheri_setboundsexact`.

Reply:
Hmm, there are definitely some existing `cheri_setboundsexact` in the kernel that don't CRRL the length. :(  (Some of the ones I added in exec recently)

Reply:
Hmm, all of the ones I added should be fine though (small enough that CRRL should be identity).  I'll fix this one.